pkg/controller/common/common.go

```diff
@@ -123,18 +123,46 @@ func (controllerContext ControllerContext) Stop() {
 	controllerContext.Broadcaster.Shutdown()
 }
 
-type ChildMap map[string]map[string]*unstructured.Unstructured
+type GroupVersionKind struct {
+	schema.GroupVersionKind
+}
+
+func (gvk GroupVersionKind) MarshalText() ([]byte, error) {
+	var marshalledText string
+	if gvk.Group == "" {
+		marshalledText = fmt.Sprintf("%s.%s", gvk.Kind, gvk.Version)
+	} else {
+		marshalledText = fmt.Sprintf("%s.%s/%s", gvk.Kind, gvk.Group, gvk.Version)
+	}
+	return []byte(marshalledText), nil
```

Comment:
Do we foresee the need to have error in the response?
Can we avoid it?

Reply:
the error here is just because of signature of `encoding.TextMarshaller` method